aws/ec2.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"fmt"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// Build string that'll be shown in resources list
+func buildEntryName(instance ec2.Instance) string {
+	return fmt.Sprintf(
+		"ec2-%s-%s",
+		awsgo.StringValue(instance.InstanceId),
+		awsgo.StringValue(instance.InstanceType),
+	)
+}
+
+// Returns a formatted string of EC2 instance ids
+func getAllEc2Instances(session *session.Session, region string) ([]*string, error) {
+	svc := ec2.New(session)
+
+	params := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: awsgo.String("instance-state-name"),
+				Values: []*string{
+					awsgo.String("running"), awsgo.String("pending"),
+					awsgo.String("stopped"), awsgo.String("stopping"),
+				},
+			},
+		},
+	}
+
+	output, err := svc.DescribeInstances(params)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	var instanceIds []*string
```

Comment:
Why an array of pointers? Why not just an array of plain strings?

Reply:
The call to terminate instances uses an array of string pointers. Easier to just get them here as opposed to trying to build the pointer array from a plain string array

Reply:
I recommend optimizing the code for readability and maintenance. It's easier to work with an array of plain strings and it communicates the intent better; an array of pointers makes it sound like some of them could be `nil` or you should be modifying them for some reason, neither of which is true.

It's very easy to turn an array of strings into an array of string pointers for the AWS API call: just wrap it with [aws.StringSlice](https://docs.aws.amazon.com/sdk-for-go/api/aws/#StringSlice).